acm/acmstate/state.go

```diff
@@ -45,6 +107,16 @@ type StorageIterable interface {
 	IterateStorage(address crypto.Address, consumer func(key binary.Word256, value []byte) error) (err error)
 }
 
+type MetadataGetter interface {
+	// Get an Metadata by its hash. This is content-addressed
+	GetMetadata(metahash MetadataHash) (string, error)
```

Comment:
I don't think this ever returns an error...

Reply:
That's a bad implementation.